### PR TITLE
Windows support for Electron

### PR DESCRIPTION
**Pull request by ldez**
Related to https://github.com/electron/electron/issues/2033
Fix https://github.com/asciidoctor/atom-asciidoc-preview/issues/159

I don't know how to write specs for this case.

### CODE REVIEW

**Comment by Mogztter** [comment]
> I don't know how to write specs for this case.

Indeed this is not a simple task and we will just test our implementation anyway :neutral_face:

File: lib/asciidoctor/opal_ext.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 %x(
   var isNode = typeof module !== 'undefined' && module.exports,
+      isElectron = process.versions !== undefined && process.versions.electron !== undefined,
```
Comment (Mogztter) [inline lib/asciidoctor/opal_ext.rb]: maybe we should check that `process` is defined ?

Comment (ldez) [inline lib/asciidoctor/opal_ext.rb]: [process](https://nodejs.org/api/process.html) is always defined with Node

Comment (ldez) [inline lib/asciidoctor/opal_ext.rb]: oops `isXX` are not functions, I must test if `process` is defined.

**Comment by Mogztter** [comment]
@mojavelinux I don't remember if we agreed on a naming to define the runtime environment ?

Electron should be:
```javascript
JAVASCRIPT_ENGINE=electron
JAVASCRIPT_PLATFORM=node // or maybe nodejs ?
```

**Comment by ldez** [comment]
`JAVASCRIPT_ENGINE` is V8 with Node.
Electron is a context perhaps `JAVASCRIPT_CONTEXT`

**Comment by Mogztter** [comment]
Yes you are right, that was part of the [discussion](https://github.com/asciidoctor/asciidoctor/issues/897#issuecomment-63955096) but I think we "formalize" something (the problem is that I can't remember where)

**Comment by Mogztter** [comment]
I think we can track this improvement in another issue

**Comment by Mogztter** [comment]
:+1: Thanks @ldez

**Pull request merged**